arch/riscv/core/smp.c

```diff
@@ -84,7 +84,11 @@ void arch_secondary_cpu_init(int hartid)
 
 #ifdef CONFIG_SMP
 
-#define MSIP_BASE 0x2000000UL
+#define CLINT_NODE DT_NODELABEL(clint)
+#if !DT_NODE_EXISTS(CLINT_NODE)
+#error "Label 'clint' is not defined in the devicetree."
```

Comment:
I'm not sure about this change. While currently there are no SMP targets in Zephyr that don't have CLINT, this could change in the future. 

I'm aware of at least one out-of-tree (but maintained) port to Linux that doesn't use CLINT - VexRiscv SMP. And there's at least one effort that I'm aware of of porting this target to Zephyr (we already have the non-SMP configuration of it in-tree): https://github.com/zephyrproject-rtos/zephyr/discussions/78053.

Reply:
Update: OK, it seems like it is already coupled and https://github.com/zephyrproject-rtos/zephyr/pull/81256 sees to decouple SMP from CLINT